### PR TITLE
Category lifecycle hooks

### PR DESCRIPTION
**Pull request by kpanayotova**
This commit adds filter and action hooks for category lifecycle events (create, update, delete).

There are different use cases for category hooks - modify category slug, auto assign image, submit category name and description to a Solr index for searching, notify plugins managing their own sorted sets etc.

### CODE REVIEW

**Comment by julianlam** [comment]
Hi @kpanayotova, please sign the [Contributor Assignment Agreement](https://docs.google.com/forms/d/1KrniXs3pwK8dMeAZnnQz2D07ne7LjJTYIYwJrpDqgnE/viewform) if you haven't already :smile:

File: src/categories/create.js [hunk 1]
```diff
@@ -39,15 +40,18 @@ module.exports = function(Categories) {
 
 				var defaultPrivileges = ['find', 'read', 'topics:create', 'topics:reply'];
 
-				async.series([
```
Comment (julianlam) [inline src/categories/create.js]: Is there a reason this is wrapped around the `async.series` instead of being one of the function calls inside?

Comment (kpanayotova) [inline src/categories/create.js]: Yes, this ensures that the series is executed over the modified category data returned by the filter.

Comment (julianlam) [inline src/categories/create.js]: Ah, right. Looks good to me, then.

**Comment by kpanayotova** [comment]
Hi @julianlam, signed already.

**Comment by barisusakli** [comment]
https://github.com/NodeBB/NodeBB/commit/dff1a04585d550906f404f361274997dc4bf0f4a

**Pull request closed without merging**